### PR TITLE
Core Media Management

### PR DESCRIPTION
**Pull request by IvanZugec**
Added the ability to upload media assets using the core Media module. 

## Functionality

- Created 5 new media types
- Converted existing Attachments and Featured image fields to Media fields
- Users can bulk upload via `admin/content/media/bulk-upload`
- Users can embed assets via entity embed

## Media type

The following media types will be created:
- Image
- File (for documents)
- Video (uploading MP4 files)
- Audio (used for uploading mp3 files)
- Remote video (used for embedding YouTube or Vimeo videos)

### CODE REVIEW

File: modules/custom/core/govcms_media/config/install/entity_browser.browser.govcms_media_file_browser.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    label: Library
+    id: view
+  0a72db77-98a3-4f5b-9168-8b502e6fa942:
+    settings:
+      form_mode: summary
+      media_type: file
+      upload_location: 'public://[date:custom:Y]-[date:custom:m]'
+      dropzone_description: 'Drop files here to upload them'
+      max_filesize: 32M
+      extensions: 'jpg jpeg gif png txt doc docx xls pdf ppt pps odt ods odp'
```
Comment (tobybellwood) [inline modules/custom/core/govcms_media/config/install/entity_browser.browser.govcms_media_file_browser.yml]: Do we need to ensure that the list of filetypes in
entity_browser.browser.govcms_media_file_browser.yml and 
entity_browser.browser.govcms_media_image_browser.yml
are matched in
field.field.media.file.field_media_file.yml and
field.field.media.image.field_media_image.yml

It should probably include csv xlsx and pptx at the minimum - but reference our current list at https://github.com/govCMS/govCMS/blob/43606d89a597c7cb38341a956ed799d02e00890a/modules/custom/govcms_core/govcms_core.install#L27

File: modules/custom/core/govcms_media/config/install/entity_browser.browser.govcms_media_image_browser.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    label: Library
+    id: view
+  da20b711-c933-403e-a9bd-8e9bd0889a8b:
+    settings:
+      form_mode: summary
+      media_type: image
+      upload_location: 'public://[date:custom:Y]-[date:custom:m]'
+      dropzone_description: 'Drop files here to upload them'
+      max_filesize: 32M
+      extensions: 'jpg jpeg gif png txt doc xls pdf ppt pps odt ods odp'
```
Comment (tobybellwood) [inline modules/custom/core/govcms_media/config/install/entity_browser.browser.govcms_media_image_browser.yml]: Do we need to ensure that the list of filetypes in
entity_browser.browser.govcms_media_file_browser.yml and 
entity_browser.browser.govcms_media_image_browser.yml
are matched in
field.field.media.file.field_media_file.yml and
field.field.media.image.field_media_image.yml

It should probably include csv xlsx and pptx at the minimum - but reference our current list at https://github.com/govCMS/govCMS/blob/43606d89a597c7cb38341a956ed799d02e00890a/modules/custom/govcms_core/govcms_core.install#L27

File: modules/custom/core/govcms_media/config/install/field.field.media.file.field_media_file.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+entity_type: media
+bundle: file
+label: File
+description: ''
+required: true
+translatable: true
+default_value: {  }
+default_value_callback: ''
+settings:
+  file_extensions: 'txt doc docx pdf'
```
Comment (tobybellwood) [inline modules/custom/core/govcms_media/config/install/field.field.media.file.field_media_file.yml]: Do we need to ensure that the list of filetypes in
entity_browser.browser.govcms_media_file_browser.yml and 
entity_browser.browser.govcms_media_image_browser.yml
are matched in
field.field.media.file.field_media_file.yml and
field.field.media.image.field_media_image.yml

It should probably include csv xlsx and pptx at the minimum - but reference our current list at https://github.com/govCMS/govCMS/blob/43606d89a597c7cb38341a956ed799d02e00890a/modules/custom/govcms_core/govcms_core.install#L27

Comment (IvanZugec) [reply modules/custom/core/govcms_media/config/install/field.field.media.file.field_media_file.yml]: @tobybellwood 
based from this list: https://github.com/govCMS/govCMS/blob/43606d89a597c7cb38341a956ed799d02e00890a/modules/custom/govcms_core/govcms_core.install#L27

Below is a breakdown of the extensions and their media type.

image: jpg jpeg gif png svg webp
file: txt doc docx xls xlsx xlsm pdf ppt pptx pps ppsx odt ods odp csv rtf
audio: mp3 ogg weba oga
video: mov mp4 m4a m4v mpeg avi ogv webm

**Comment by IvanZugec** [comment]
@tobybellwood Fixed the file extensions as requested in 4cce73b

**Comment by IvanZugec** [comment]
Merged the latest from 1.x branch.

**Comment by IvanZugec** [comment]
Added the ability to tag media types and filter by tags.

**Review by pandaskii — APPROVED** [review]
From my point of view, this PR is ready for merging. We can leave the tweaking and features to next stage

**Comment by tobybellwood** [comment]
@pandaskii you want to merge it then - or I can push the buttons if needed 😈

**Pull request merged**